applications/audio/broadcast_config_tool/src/main.c

```diff
@@ -1004,6 +1119,511 @@ static int cmd_location(const struct shell *shell, size_t argc, char **argv)
+
+	int encrypt = strtoul(argv[1], NULL, 10);
+
+	if (encrypt != 1 && encrypt != 0) {
+		shell_error(shell, "Encryption must be 0 or 1");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[2])) {
+		shell_error(shell, "BIG index must be a digit");
+		return -EINVAL;
+	}
+
+	uint8_t big_index = (uint8_t)atoi(argv[2]);
+
+	if (big_index >= CONFIG_BT_ISO_MAX_BIG) {
+		shell_error(shell, "BIG index out of range");
+		return -EINVAL;
+	}
+
+	broadcast_param[big_index].encryption = encrypt;
+
+	if (encrypt == 1) {
+		if (argc < 4) {
+			shell_error(shell, "Broadcast code must be set");
+			return -EINVAL;
+		}
+
+		if (strlen(argv[3]) > BT_AUDIO_BROADCAST_CODE_SIZE) {
+			shell_error(shell, "Broadcast code must be %d characters long",
+				    BT_AUDIO_BROADCAST_CODE_SIZE);
+			return -EINVAL;
+		}
+		memset(broadcast_param[big_index].broadcast_code, '\0',
+		       ARRAY_SIZE(broadcast_param[big_index].broadcast_code));
+		memcpy(broadcast_param[big_index].broadcast_code, argv[3],
+		       BT_AUDIO_BROADCAST_CODE_SIZE);
+	}
+
+	return 0;
+}
+
+static int cmd_adv_name(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 2) {
+		shell_error(shell, "Usage: bct device_name <name> <BIG index>");
+		return -EINVAL;
+	}
+
+	if (strlen(argv[1]) > CONFIG_BT_DEVICE_NAME_MAX) {
+		shell_error(shell, "Device name too long");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[2])) {
+		shell_error(shell, "BIG index must be a digit");
+		return -EINVAL;
+	}
+
+	uint8_t big_index = (uint8_t)atoi(argv[2]);
+
+	if (big_index >= CONFIG_BT_ISO_MAX_BIG) {
+		shell_error(shell, "BIG index out of range");
+		return -EINVAL;
+	}
+
+	memset(broadcast_param[big_index].adv_name, '\0',
+	       ARRAY_SIZE(broadcast_param[big_index].adv_name));
+	memcpy(broadcast_param[big_index].adv_name, argv[1], strlen(argv[1]));
+
+	return 0;
+}
+
+static int cmd_program_info(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc != 4) {
+		shell_error(shell, "Usage: bct program_info \"info\" <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[2])) {
+		shell_error(shell, "BIG index must be a digit");
+		return -EINVAL;
+	}
+
+	uint8_t big_index = (uint8_t)atoi(argv[2]);
+
+	if (big_index >= CONFIG_BT_ISO_MAX_BIG) {
+		shell_error(shell, "BIG index out of range");
+		return -EINVAL;
+	}
+
+	if (!is_number(argv[3])) {
+		shell_error(shell, "Subgroup index must be a digit");
+		return -EINVAL;
+	}
+
+	uint8_t sub_index = (uint8_t)atoi(argv[3]);
+
+	if ((sub_index >= CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT) ||
+	    (broadcast_param[big_index].subgroups == NULL) ||
+	    (sub_index >= broadcast_param[big_index].num_subgroups)) {
+		shell_error(shell, "Subgroup index out of range");
+		return -EINVAL;
+	}
+
+	bt_audio_codec_cfg_meta_set_program_info(
+		&broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.codec_cfg,
+		argv[1], strlen(argv[1]));
+
+	return 0;
+}
+
+static int usecase_find(const struct shell *shell, const char *name)
```

Comment:
```suggestion
static enum usecases usecase_find(const struct shell *shell, const char *name)
```

Reply:
This returns -ESRCH if the usecase is not found

Reply:
Ah true